streams/src/main/java/org/apache/kafka/streams/processor/api/AbstractProcessor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.api;
+
+/**
+ * An abstract implementation of {@link Processor} that manages the {@link ProcessorContext} instance and provides default no-op
+ * implementation of {@link #close()}.
+ *
+ * @param <KIn> the type of input keys
+ * @param <VIn> the type of input values
+ * @param <KOut> the type of output keys
+ * @param <VOut> the type of output values
+ */
+public abstract class AbstractProcessor<KIn, VIn, KOut, VOut> implements Processor<KIn, VIn, KOut, VOut> {
```

Comment:
One minor thing that bugged me about the old PAPI was the AbstractProcessor class. It was widely used in places where it provided no value. I'm worried that we will just perpetuate here.

Since the new API has default implementations for init and close, the only value this new abstract class provides is when the subclass needs the context. In that case, we save them from the boilerplace of saving off the context in a field.

Therefore, I'd like to call this class `ContextualProcessor` instead of `AbstractProcessor`. This should make it a little harder to depend on this class unnecessarily.

```suggestion
public abstract class ContextualProcessor<KIn, VIn, KOut, VOut> implements Processor<KIn, VIn, KOut, VOut> {
```

Reply:
Really like this suggestion. Thanks @vvcephei!